### PR TITLE
feat: adds separate UI component module using @atb-as/theme

### PR DESCRIPTION
**Pull request by mikaelbr**
~_This is kind of dirty PR, I apologize. My elm format doesn't seem to match yours @mjenssen making the changeset bigger and harder to follow._~

This PR introduces UI module with components for section box and status message and starts a system for making reusable components. Currently just used in order summary.

Screenshot 2021-04-12 at 09 47 43

### CODE REVIEW

File: src/elm/Ui/Section.elm [hunk 1]
```diff
@@ -0,0 +1,43 @@
+module Ui.Section exposing (..)
```
Comment (mjenssen) [inline src/elm/Ui/Section.elm:1]: Only expose the symbols that need exposing. Even if this means everything in the file/module, it's nice as a documentation of "this is the stuff in this module".

File: package.json [hunk 2]
```diff
@@ -45,7 +45,9 @@
     "webpack-subresource-integrity": "1.4.0",
     "workbox-webpack-plugin": "5.0.0-rc.2"
   },
-  "dependencies": {},
+  "dependencies": {
+    "@atb-as/theme": "^1.3.0"
```
Comment (mjenssen) [inline package.json:49]: Shouldn't this be a dev-dependency? (it's not needed while running the final generated code)

**Review by mjenssen — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (mikaelbr) [reply package.json:49]: Hmm. Hmm. Somewhat unsure what I think here. Yes, technically not used in runtime as it is bundled down. But the way I think dependencies vs devDependencies is most clear when talking about CDN hosted solutions is "is this used as a part of the output or to the user" -> dependency. "is this used internally as a tool or when developing" -> dev dependency. That communicates clearer (to me) what kind of changes and testing needs to be done when managing dependencies (updating, removing, analyzing etc).

Comment (mjenssen) [reply package.json:49]: dep vs dev-dep makes more sense when using node to run code. In that case, dep contains all the dependencies that are needed to run the code, while dev-dep only contains dependencies that are needed while developing (e.g. testing, formatting, code generation, etc.). In frontend projects you aren't really running anything (unless it's node with SSR), so you end up not having any normal dependencies. That's how I've been thinking about this at least.

Comment (mikaelbr) [reply package.json:49]: Not a hill I'm prepared to die on, so I can change 😬 . What you said is the exact same reasoning as mine I think. With frontend projects, I operate with the definition as I mentioned above to get added semantical benefit of separating devDeps and normal deps.

Main reason for having deps is: reduce install on production (not applicable), and explicit relations if used as a dependency itself (not applicable). So here it ends up with no actual difference other than with the semantics we define it as.

**Review by mjenssen — APPROVED** [review]

**Pull request merged**